### PR TITLE
Blog App: Data in models

### PR DESCRIPTION
**Pull request by Mike111222**
- Create models

Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png).

- Set up associations between models.

Remember that author_id column in posts table should be the foreign_key for the users table.

- Use models to insert data

- Familiarize yourself with ['rails c' command](https://guides.rubyonrails.org/command_line.html#bin-rails-console).

- Open rails console for your project.
- 

- Create at least one user by running the following code:

first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')

- Create at least 4 posts written by one of the users you created by running the following code:

first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post)

- Create at least 6 posts comments for one of the posts you created by running the following code:

Comment.create(post: first_post, author: second_user, text: 'Hi Tom!' )
Use other [CRUD methods](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/active_record_crud.md) to find, update and delete entities.

- Create custom methods

Your models should include some additional methods.

- Users

A method that returns the 3 most recent posts for a given user.

- Posts

A method that updates the posts counter for a user.
A method which returns the 5 most recent comments for a given post.

- Comments

A method that updates the comments counter for a post.

- Likes 

A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,2 +1,17 @@
...
+    posts.order(created_at: :desc).limit(3)
+  end
+
+  def update_posts_counter
+    update(posts_counter: posts.count)
+  end
+
+  def update_comments_counter
+    update(comments_counter: comments.count)
+  end
```
Comment (julie-ify) [inline app/models/user.rb]: - [x] The user model does not have a comments_counter column so there is no use in having this method here. Kindly remove it.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Mike111222,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (Mike111222) [reply app/models/user.rb]: I have made the changes you mentioned. Thanks a lot for your kind feedback.

**Review by Gambit142 — APPROVED** [review]
Hi @Mike111222 👋, 

## Status: Approved 🟢   

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

### Highlights
- Setup linters correctly ✔️ 
- Created a file for the `User` model and its methods ✔️ 
- Created a file for the `Post` model and its methods ✔️ 
- Created a file for the `Comment` model and its methods ✔️ 
- Created a file for the `Like` model and its methods ✔️ 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**